js/test/Exchange/test.ledgerItem.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+// ----------------------------------------------------------------------------
+
+const log = require ('ololog')
+    , ansi = require ('ansicolor').nice
+    , chai = require ('chai')
+    , expect = chai.expect
+    , assert = chai.assert
+
+/*  ------------------------------------------------------------------------ */
+
+module.exports = (exchange, item, code, now) => {
+    expect (item).to.be.an ('object')
+    expect (item).to.have.property ('id')
+    expect (typeof item.id).to.be.oneOf (['string', 'undefined'])
+    expect (item).to.have.property ('direction')
+    expect (item.direction).to.be.oneOf (['in', 'out'])
+    expect (item).to.have.property ('account')
+    expect (typeof item.account).to.be.oneOf (['string', 'undefined'])
+    expect (item).to.have.property ('referenceId')
+    expect (typeof item.referenceId).to.be.oneOf (['string', 'undefined'])
+    expect (item).to.have.property ('referenceAccount')
+    expect (typeof item.referenceAccount).to.be.oneOf (['string', 'undefined'])
+    expect (item).to.have.property ('type')
+    expect (item.type).to.be.oneOf (['trade', 'transaction', 'margin', 'cashback', 'referral', 'transfer', 'fee', /* TODO: add more types here */ ])
+    expect (item).to.have.property ('currency')
+    expect (exchange.currencies).to.have.property (item.currency)
+    expect (item).to.have.property ('amount')
+    expect (typeof item.amount).to.be.oneOf (['number', 'undefined'])
+    expect (item).to.have.property ('balance')
```

Comment:
Yes, `balance` was not there. But since exchange provides it, i think it's reasonable to keep it here. It would be very useful for the balance history. The property represents balance after the operation was applied.

Reply:
@tankakatan i suggest that we add `balanceBefore` and `balanceAfter` then, because, I think, we will meet exchanges with both or either of the two.

Reply:
> @tankakatan i suggest that we add `balanceBefore` and `balanceAfter` then, because, I think, we will meet exchanges with both or either of the two.

Will do.